File: LoanManagementApp02/LoanManagementApp02/LoanManagementApp02/Controllers/LoanController.js
// Loan management logic for applying, approving, and rejecting loans
const Loan = require('../Models/Loan');

// Apply for a Loan
exports.applyLoan = async (req, res) => {
    const { user_id, amount } = req.body;

    const loan = new Loan({
        user_id,
        amount,
        status: 'Pending'
    });

    await loan.save();
    res.status(201).json({ message: 'Loan application submitted.' });
};

// Approve Loan
exports.approveLoan = async (req, res) => {
    const { id } = req.params;
    await Loan.findByIdAndUpdate(id, { status: 'Approved' });
    res.status(200).json({ message: 'Loan approved.' });
};

// Reject Loan
exports.rejectLoan = async (req, res) => {
    const { id } = req.params;
    await Loan.findByIdAndUpdate(id, { status: 'Rejected' });
    res.status(200).json({ message: 'Loan rejected.' });
};

File: LoanManagementApp02/LoanManagementApp02/LoanManagementApp02/Controllers/LoanController.cs
using Fluent.Infrastructure.FluentModel;
using LoanManagementSystem.Models;
using Microsoft.AspNetCore.Mvc;

namespace LoanManagementSystem.Controllers
{
    [ApiController]
    [Route("api/loans")]
    public class LoanController : ControllerBase
    {
        private readonly ApplicationDbContext _context;

        public LoanController(ApplicationDbContext context)
        {
            _context = context;
        }

        [HttpPost("apply")]
        public async Task<IActionResult> Apply([FromBody] Loan loan)
        {
            try
            {
                loan.Id = Guid.NewGuid();
                loan.CreatedAt = DateTime.UtcNow;
                _context.Loan.Add(loan);
                await _context.SaveChangesAsync();

                return Ok("Loan application submitted successfully.");
            }
            catch (Exception ex)
            {
                // Log the exception (ex) here if needed
                return StatusCode(500, "Internal server error.");
            }
        }

        [HttpGet("{id}")]
        public async Task<IActionResult> GetLoan(Guid id)
        {
            try
            {
                var loan = await _context.Loans.Include(l => l.User).SingleOrDefaultAsync(l => l.Id == id);
                if (loan == null)
                {
                    return NotFound("Loan not found.");
                }

                return Ok(loan);
            }
            catch (Exception ex)
            {
                // Log the exception (ex) here if needed
                return StatusCode(500, "Internal server error.");
            }
        }

        [HttpPut("{id}/approve")]
        public async Task<IActionResult> ApproveLoan(Guid id)
        {
            try
            {
                var loan = await _context.Loans.SingleOrDefaultAsync(l => l.Id == id);
                if (loan == null)
                {
                    return NotFound("Loan not found.");
                }

                loan.Status = "Approved";
                await _context.SaveChangesAsync();

                return Ok("Loan approved successfully.");
            }
            catch (Exception ex)
            {
                // Log the exception (ex) here if needed
                return StatusCode(500, "Internal server error.");
            }
        }

        [HttpPut("{id}/reject")]
        public async Task<IActionResult> RejectLoan(Guid id)
        {
            try
            {
                var loan = await _context.Loans.SingleOrDefaultAsync(l => l.Id == id);
                if (loan == null)
                {
                    return NotFound("Loan not found.");
                }

                loan.Status = "Rejected";
                await _context.SaveChangesAsync();

                return Ok("Loan rejected successfully.");
            }
            catch (Exception ex)
            {
                // Log the exception (ex) here if needed
                return StatusCode(500, "Internal server error.");
            }
        }
    }
}
